tests/data/test358

```diff
@@ -0,0 +1,94 @@
+<testcase>
+<info>
+<keywords>
+HTTP
+HTTP GET
+Alt-Svc
+HTTP/2
+</keywords>
+</info>
+
+#
+# Server-side
+<reply>
+<data nocheck="yes">
+HTTP/1.1 200 OK
+Date: Thu, 09 Nov 2010 14:49:00 GMT
+Content-Length: 6
+Connection: close
+Content-Type: text/html
+Funny-head: yesyes
+Alt-Svc: h2=":9014", ma=315360000; persist=0
+
+-foo-
+</data>
+</reply>
+
+#
+# Client-side
+<client>
+<features>
+alt-svc
+debug
+</features>
+<server>
+http
+http/2
+</server>
+ <name>
+HTTPS GET translated by alt-svc lookup to HTTP/2 GET
+ </name>
+<setenv>
+# make debug-curl accept Alt-Svc over plain HTTP
+CURL_ALTSVC_HTTP="yeah"
+</setenv>
+ <command>
+--http2 --alt-svc "log/altsvc-358" "http://%HOSTIP:%HTTPPORT/358" "http://%HOSTIP:%HTTPPORT/358"
```

Comment:
Should we really need to have `--http2` specified to get this to work? I understand you need it now, but isn't this actually a bug you hide by using this flag? Goes for both test cases.

Reply:
yes, I was thinking (pedantically) that my bug fix would come with new PR fixing code and these tests ... happy to bundle all of em together.

Reply:
It just felt weird to land tests that verifies the wrong behavior, but you're right we can also view it as verifying *current* behavior and then change that in a subsequent bug-fix. I'll be okay with that if you prefer that way.